ee/tabby-webserver/src/oauth/mod.rs

```diff
@@ -0,0 +1,37 @@
+use std::sync::Arc;
+use axum::{Json, Router, routing};
+use axum::extract::{Query, State};
+use axum::http::StatusCode;
+use axum::middleware::from_fn_with_state;
+use serde::Deserialize;
+use tracing::error;
+use crate::oauth::github::GithubOAuthResponse;
+use crate::schema::ServiceLocator;
+
+mod github;
+
+pub fn routes(locator: Arc<dyn ServiceLocator>) -> Router {
+    Router::new()
+        .route("/github/callback", routing::get(github_callback))
+        .with_state(locator)
+}
+
+#[derive(Deserialize)]
+struct GithubCallbackParams {
+    code: String,
+    state: Option<String>,
+}
+
+async fn github_callback(
+    State(state): State<Arc<dyn ServiceLocator>>,
+    Query(params): Query<GithubCallbackParams>,
+) -> Result<Json<GithubOAuthResponse>, StatusCode> {
+    let resp = github::access_token(state, params.code)
```

Comment:
need to check whether the github oauth is actual active (return 404 when active = false)